File: backend/app/schemas/document.py
import uuid
from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import DocumentCategory


class DocumentBase(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    category: DocumentCategory
    expiry_date: Optional[date] = None
    notes: Optional[str] = None


class DocumentCreate(DocumentBase):
    pass


class DocumentUpdate(BaseModel):
    name: Optional[str] = Field(default=None, min_length=1, max_length=255)
    category: Optional[DocumentCategory] = None
    expiry_date: Optional[date] = None
    notes: Optional[str] = None


class DocumentRead(DocumentBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    file_name: Optional[str] = None
    file_size: Optional[int] = None
    mime_type: Optional[str] = None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/document.ts
export type DocumentCategory = "id" | "passport" | "license" | "insurance" | "warranty" | "contract" | "invoice" | "other"

export interface Document {
  id: string
  name: string
  category: DocumentCategory
  expiry_date: string | null
  notes: string | null
  file_name: string | null
  file_size: number | null
  mime_type: string | null
  created_at: string
  updated_at: string
}

export interface DocumentPayload {
  name: string
  category: DocumentCategory
  expiry_date: string | null
  notes: string | null
}
